pyomo/contrib/pyros/util.py

```diff
@@ -0,0 +1,683 @@
+                degree_param_powers = degree_param_powers[::-1]
+                for param_powers in degree_param_powers:
+                    product = 1
+                    for idx, power in enumerate(param_powers):
+                        if power == 0:
+                            pass
+                        else:
+                            product = product * uncertain_params[idx]**power
+                    e += Z.pop(0) * product
+            model_data.working_model.add_component("decision_rule_eqn_" + str(i),
+                                                       Constraint(expr=e == second_stage_variables[i]))
+            decision_rule_eqns.append(getattr(model_data.working_model, "decision_rule_eqn_" + str(i)))
+            if len(Z) != 0:
+                raise RuntimeError("Construction of the decision rule functions did not work correctly! "
+                                   "Did not use all coefficient terms.")
+    model_data.working_model.util.decision_rule_eqns = decision_rule_eqns
+    return
+
+
+def identify_objective_functions(model, config):
+    '''
+    Determine the objective first- and second-stage costs based on the user provided variable partition
+    :param model: deterministic model
+    :param config: config block
+    :return:
+    '''
+
+    m = model
+    obj = [o for o in model.component_data_objects(Objective)]
+    if len(obj) > 1:
+        raise AttributeError("Deterministic model must only have 1 active objective!")
+    if obj[0].sense != minimize:
+        raise AttributeError("PyROS requires deterministic models to have an objective function with  'sense'=minimization. "
+                             "Please specity your objective function as minimization.")
+    first_stage_terms = []
+    second_stage_terms = []
+
+    first_stage_cost_expr = 0
+    second_stage_cost_expr = 0
+    const_obj_expr = 0
+
+    if isinstance(obj[0].expr, Var):
+        obj_to_parse = [obj[0].expr]
+    else:
+        obj_to_parse = obj[0].expr.args
+    first_stage_variable_set = ComponentSet(model.util.first_stage_variables)
+    second_stage_variable_set = ComponentSet(model.util.second_stage_variables)
+    for term in obj_to_parse:
+        vars_in_term = list(v for v in identify_variables(term))
+
+        first_stage_vars_in_term = list(v for v in vars_in_term if
+                                        v in first_stage_variable_set)
+        second_stage_vars_in_term = list(v for v in vars_in_term if
+                                         v in second_stage_variable_set)
+        for v in first_stage_vars_in_term:
+            if id(v) not in list(id(var) for var in first_stage_terms):
+                first_stage_terms.append(v)
+        for v in second_stage_vars_in_term:
+            if id(v) not in list(id(var) for var in second_stage_terms):
+                second_stage_terms.append(v)
+
+        if len(first_stage_vars_in_term) > 0 and len(second_stage_vars_in_term) > 0:
```

Comment:
You can simplify this to
```python
if first_stage_vars_in_term and second_stage_vars_in_term:
```
(similar simplifications also apply to the next three clauses in the if ladder.)